### ISSUE TITLE
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire on my iPhone(iOS8) while debuging

### ISSUE THREAD

**Issue by MikeChongCan** [Alamofire/Alamofire#101]
```
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /private/var/mobile/Containers/Bundle/Application/XXX/Anachronos.app/Anachronos
  Reason: image not found
```

Xcode6 GM

I've read all the issues about this kind of problem, but couldn't solve the problem.

Also, there is the same issue in  iOS Example project:

```
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /private/var/mobile/Containers/Bundle/Application/CEA6C4C1-C211-460C-8ECF-4FD9C66F1016/iOS Example.app/iOS Example
  Reason: image not found
```

**Comment by hernanarber** [comment]
THANKS :-)

**Comment by KevinIsSun** [comment]
I have the same question, I just do as Mikoj told, but i still crash on my iPhone, not in simulator.I really do not know how to do

**Comment by minhoryang** [comment]
Me too as KevinIsSun said.. Is this really resolved?

**Comment by mmmattos** [comment]
Thanks all for the insights in solving this issue.
Just one clarification. The "Embed Frameworks" option was not available on xCode 6.3 under Build Phases. It is actually a "Copy Files Phase" (clicking in the +) in the same Build Phases.
I had to do it too for other framework I use (SwiftyJSON).

**Comment by mushu8** [comment]
I have the same problem except that my Alamofire is embeded in a framework that is linked to my project…
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /private/var/mobile/Containers/Bundle/Application/4D3EC3B3-70DF-44A1-878D-03D717C65CFD/Fabville Recette.app/Fabville Recette
  Reason: image not found

**Comment by juanfv2** [comment]
Thank you very much!

**Comment by Nadohs** [comment]
@Mikoj if this were SO, I would upvote you. thanks

**Comment by tomaszwojcik** [comment]
Still have the same issue, XCode 6.4, iOS 8.0+, Swift 1.2, using CocoaPods, Alamofire 1.3.

```
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /private/var/mobile/Containers/Bundle/Application/76F222DA-1278-463E-976B-AA9C3497625E/XXX.app/XXX
  Reason: no suitable image found.  Did find:
	/private/var/mobile/Containers/Bundle/Application/76F222DA-1278-463E-976B-AA9C3497625E/RemoteMyApp.app/Frameworks/Alamofire.framework/Alamofire: mach-o, but wrong architecture
	/private/var/mobile/Containers/Bundle/Application/76F222DA-1278-463E-976B-AA9C3497625E/RemoteMyApp.app/Frameworks/Alamofire.framework/Alamofire: mach-o, but wrong architecture
	/private/var/mobile/Containers/Bundle/Application/76F222DA-1278-463E-976B-AA9C3497625E/RemoteMyApp.app/Frameworks/Alamofire.framework/Alamofire: mach-o, but wrong architecture

**Comment by kevin-hirsch** [comment]
@Mikoj It worked. Thank you!

**Comment by meigesir** [comment]
@Mikoj It worked too. Thank you!

**Comment by shihkai-tony-wei** [comment]
In the Build Phases -> copy  phrase,
it says  build/Debug-iphoneos. does this mean it is using the library built with debug configuration in my release app?

**Comment by mikecamara** [comment]
@Mikoj It worked. Thank you! :+1:

**Comment by BellAppLab** [comment]
+1 @Mikoj 

Closed to prevent too many "thank you" and "you're awesome".

**Comment by nelsongc** [comment]
it working, thaks

**Comment by derpoliuk** [comment]
@mushu8 Did you resolve this issue with Alamofire inside framework?

**Comment by abury** [comment]
@derpoliuk @mushu8 Came across this thread trying to solve a similar problem when adding a 3rd party framework to a framework i'm building.
I resolved it by heading to build phases and adding a 'new copy files phase'. Set destination to 'Frameworks' and then drag the framework into the list

(Example shows SDWebImage framework, but has universally solved the same problem I was having with Realm and MockingJay)
<img width="1200" alt="screen shot 2015-12-28 at 5 12 40 pm" src="https://cloud.githubusercontent.com/assets/982276/12016477/3b70728a-ad86-11e5-8b83-503889c864db.png">

Hope it helps

**Comment by comlog** [comment]
I have the same problem ,It does not work ,how to dear the problem? for help online

**Comment by korovyev** [comment]
@abury what that method produces in the client of your frameworks project is this:
<img width="305" alt="screen shot 2016-01-02 at 17 58 21" src="https://cloud.githubusercontent.com/assets/1238462/12076495/cf68c31e-b17a-11e5-930f-bef39b754b6c.png">

Unfortunately they will be able to see every framework you use in your own framework. I believe the correct way to do this (correct me if I'm wrong) is to get the client to include each framework themselves as all frameworks should be linked and copied app-level. It reduces problems with versioning of frameworks also I believe.

What I would like to do is embed frameworks inside my framework, and not have them accessible by, or even visible to, the user of my framework. Unfortunately I think that is not possible, so I am starting to think I cannot use Alamofire in my framework.

**Comment by abury** [comment]
@korovyev You're right in that it shows that the frameworks can be seen from within the framework itself (I didn't know it did that!) But I'm not sure that they can then be used in the host application.
I just tried to import one of those frameworks into the host application and it didn't recognise it. Only when I imported that framework into the host app itself did it work ok, so you shouldn't run into any issues with versioning.
Whether or not the client app should include dependancies themselves I think depends on how your designing the framework itself to be built. I want something that's simple for people to start using straight away without worrying about managing dependancies. While there may be a small duplication of code if the host app also chooses to use the same framework, it's a matter of < 500 kb (as well as being hypothetical)
Alternatively you can just include the raw source files and see if that works :smile:

**Comment by iparq** [comment]
![monosnap 2016-01-05 15-47-14](https://cloud.githubusercontent.com/assets/5027726/12130968/a182affa-b3c3-11e5-8358-3b61dda55448.jpg)

Great.  WHICH ONE?

**Comment by abury** [comment]
@iparq Lol, not sure why you have so many there. I actually can't recommend that approach anymore. I've discovered that Apple actually rejects apps the include frameworks with embedded frameworks. Not idea why and it seems a bit counter intuitive, but thats the current way of it.
I ended up just including the source files of the frameworks I wanted into the framework. Not the cleanest (and gets even worse if they are Obj-C files as you have no bridging header file and have to use .h umbrella file which exposes them classes to the parent app) but that's literally the only solution for now.

**Comment by delctrl** [comment]
Still getting the same issue on iOS 9.1, with Alamofire 3.0.1. Tried some solutions found both on this page and on SO, but nothing seems to solve it

**Comment by troydo42** [comment]
Run into the same issue. It works on my device but then throw the error when i run it in the Simulator.
![screen shot 2016-05-19 at 7 40 25 pm](https://cloud.githubusercontent.com/assets/1859661/15416400/e20e5dd4-1dff-11e6-8ad8-6d3795ae4ea7.png)

**Comment by bryanbartow** [comment]
@troydo42 Same here...

**Comment by i3omar** [comment]
Had the same issue and finally solved it.
The app would run on the simulator but not my iPhone. It drove me crazy. After a few hours I went to the Keychain Access and realized that my certificate says (this certificate was signed by an unknown authority).

![cert](https://cloud.githubusercontent.com/assets/7313080/15700505/0b452aec-2789-11e6-8334-0b2ca5435c9e.png)

Solved this by following the steps [here](http://stackoverflow.com/questions/13820680/this-certificate-was-signed-by-an-unknown-authority).

I ran the app on my iPhone after that and it worked!

**Comment by nissaba** [comment]
Me I am getting this 

```
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /private/var/containers/Bundle/Application/46774FDD-2C90-41A8-8BBA-EF0F34F37DD1/AppTest.app/Frameworks/SDKTest.framework/SDKTest
  Reason: no suitable image found.  Did find:
	/private/var/containers/Bundle/Application/46774FDD-2C90-41A8-8BBA-EF0F34F37DD1/AppTest.app/Frameworks/SDKTest.framework/Frameworks/Alamofire.framework/Alamofire: mmap() error 1 at address=0x00783000, size=0x00064000 segment=__TEXT in Segment::map() mapping /private/var/containers/Bundle/Application/46774FDD-2C90-41A8-8BBA-EF0F34F37DD1/AppTest.app/Frameworks/SDKTest.framework/Frameworks/Alamofire.framework/Alamofire
```

**Comment by wilferrel** [comment]
@i3omar  After a full day of trying different things this was the missing thing that got me going. THANKS!

**Comment by gmogames** [comment]
What worked for me was this comment:
"Is there the Embed Frameworks build phase? If there isn't, you should add a new "Copy Files Phase", then select "Frameworks" for Destination, add frameworks to the list, and check "Code Sign On Copy" checkboxes."

**Comment by Legendgp** [comment]
you are good!

**Comment by Legendgp** [comment]
thanks very mach!

**Comment by i3omar** [comment]
@wilferrel Glad it helped!

**Comment by madhumitha89** [comment]
I also have the same issue.I tried every solutions above.But not working.Please help me to solve this

**Comment by iosraven** [comment]
I have tried.  And it could build..   u are very helpful.   thanks!  @Mikoj

**Comment by binjCN** [comment]
This issue also showed up for me ,please help me！

**Comment by madhumitha89** [comment]
Hi,

Embed Pods framework is the reason for this.UITest -> Buildphases -> Embed
Pods framework is missing.Copy from target and paste it in UITest target.It
worked for me.

**Comment by thanhdevapp** [comment]
Project -> General Tab -> Linked Framworks and Libraries -> Choose AlamorefireImage.framework

**Comment by adammendoza** [comment]
When you have this:
Project -> General Tab -> Linked Framworks and Libraries -> Choose AlamorefireImage.framework
Should the Status be "Required" or "Optional"?

**Comment by adammendoza** [comment]
Besides Linked Framework and Embed, I also had to add a Copy Phase

**Comment by gurumobile** [comment]
I resolve this issue as like followings.
In General->Embedded Binaries, I added necessary framework.
So app is working as well.
Thanks.

**Comment by troyibm** [comment]
Even after doing what was suggested by Mikoj I was still getting this error.  It wasn't until I set 'enable bitcode' to No in build settings that it started to work.  :/

**Comment by grantkemp** [comment]
Hi guys- I noticed that loads of people are still getting this error. 
I have found a fix that has worked for me and is pretty easy. 
I think its down to permissions.  I have logged the issue with Apple as they really need to fix this bug. 

When I changed the permission of the world Wide Developer cert to `Use System defaults` then it worked. I have included the steps and screenshots in the link below 

I would encourage you to log the ticket in apple bug report as mentioned here as Apple really should solve this massive error: 
[link to SO fix steps ](http://stackoverflow.com/a/41401354/559760)

**Comment by NrmeenTomoum** [comment]
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /var/containers/Bundle/Application/CC9B5DD0-5243-4044-8798-304DE53EE417/aswaq.app/aswaq
  Reason: no suitable image found.

**Comment by peixuan197** [comment]
Hi guys,I also get this error,and I still did not resolve after all of the methods used！Then I did not fix the problem after changed the Xcode version, **but** I did not have a problem on another Mac.So what can I do? it got me crazy!!!

**Comment by grantkemp** [comment]
Hi @peixuan197  it still happens periodically with me. I find that when I run my tests a second time - then it works ok.  
I suggest raising the issue as but with Xcode - its something that has clearly been introduced in the last two versions of xcode ( 8.21 for me) I never had this before. Compile times also seem to have gone up drastically.  

 - Log your bug here: https://bugreport.apple.com
🔥 If you reference Apple bug: 29915305 - you can help them see how big the issue is.

**Comment by like-milk** [comment]
Thank you very much!, This was very useful @Mikoj

**Comment by htarwara6245** [comment]
@OmarAlEisa , Special thanks to you, I did experienced this because I removed all certificates from my keychain!!! Thanks to you.

**Comment by daibodhisattva** [comment]
Just had the same issue on High Sierra with Xcode 9.1.  After trying the dozens and dozens of suggestions on various posts, our solution was simply to change the code signing authority from one team member to another.  We deleted the problem cert, and the app installed and ran on the test phone fine.  Reinstalled the originally signing authority profile and compiled again... all good now.  I'd recommend making this the first step in testing for this error.  We tried deleting derived data and cache, reinstalling Xcode from scratch, etc.  Nothing we did, not settings adjusted had any effect until we simply swapped out the signing profiles and ran a build.  After that, all issues resolved.

**Comment by getSekhoned** [comment]
The only thing that worked for me:
Target > Build Phases > [CP] Embed Pods Frameworks
Uncheck "Show environment variables in build log" and "Run script only when installing"

**Comment by steewsc** [comment]
Also (this worked for me) if you are using multiple configuration (Dev, Staging, Production...) check that those configurations are using proper Pods xcconfig files (see .dev, .test and .release):
<img width="1123" alt="screen shot 2017-12-27 at 4 54 40 pm" src="https://user-images.githubusercontent.com/1216079/34386244-042540f8-eb28-11e7-9ddb-deba40338791.png">

**Comment by KiranJasvanee** [comment]
cleaned Derived Data and worked! 👍 :))

**Comment by furkanvijapura** [comment]
I Have new issue faced in Alamofire Framwork

error Description >>

```
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /Users/sunilyadav/Library/Developer/CoreSimulator/Devices/76022694-CD0A-44EC-BCF7-B2470C0CF484/data/Containers/Bundle/Application/BEC17A4E-E681-44A3-A83F-6AC58601A51C/Odin_App_Project_Swift.app/Odin_App_Project_Swift
  Reason: image not found
```

plz help solve this isuue

**Comment by kaiyuanheshang** [comment]
1. deployment version do not match your min iOS version
2. you build the framework as its mach-o as dynamic library

**Comment by iOSUser110** [comment]
@grantkemp Thanks , it worked for me (y)

**Comment by swapnizzle** [comment]
I encountered this issue when trying to build to my actual iOS device (but it worked fine in the simulator). I spent at least 10 hours trying to fix this bug. Tried everything in all the forums, nothing worked. But this is what finally worked for me!

- Upgrading from iOS 13.3.1 to iOS 13.4.1

I couldn't have imagined this, but its true that <b> iOS 13.3.1 breaks the third parties signature unless you have a paid developer account!</b>
